frontend/src/hooks/api/getters/useConnectedEdges/useConnectedEdges.ts

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from 'react';
+import useUiConfig from '../useUiConfig/useUiConfig';
+import { formatApiPath } from 'utils/formatPath';
+import handleErrorResponses from '../httpErrorResponseHandler';
+import { useConditionalSWR } from '../useConditionalSWR/useConditionalSWR';
+import type { ConnectedEdge } from 'interfaces/connectedEdge';
+import type { SWRConfiguration } from 'swr';
+import { useUiFlag } from 'hooks/useUiFlag';
+
+const DEFAULT_DATA: ConnectedEdge[] = [];
+
+export const useConnectedEdges = (options?: SWRConfiguration) => {
+    const { isEnterprise } = useUiConfig();
+    const edgeObservabilityEnabled = useUiFlag('edgeObservability');
+
+    const { data, error, mutate } = useConditionalSWR<ConnectedEdge[]>(
```

Comment:
Should we use `useEnterpriseSWR` instead? It already validates is enterprise, but maybe because of the flag is not easy to use

Reply:
Unfortunately `useEnterpriseSWR` does not support a condition, so it wouldn't be practical with the flag, like you guessed. `useConditionalSWR` allows to specify the exact condition we want, which feels like the way to go in most cases. I would go as far as suggest we drop `useEnterpriseSWR` completely at some point.